test/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring.Tests/Linux/LinuxCountersTests.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics.Metrics;
+using System.IO;
+using System.Threading;
+using FluentAssertions;
+using Microsoft.Extensions.Diagnostics.ResourceMonitoring.Linux.Network;
+using Moq;
+using Xunit;
+
+namespace Microsoft.Extensions.Diagnostics.ResourceMonitoring.Linux.Test;
+
+public class LinuxCountersTests
+{
+    [Fact]
+    public void LinuxNetworkCounters_Registers_Instruments()
+    {
+        var fileSystem = new HardcodedValueFileSystem(new Dictionary<FileInfo, string>
+        {
+            {
+                new FileInfo("/proc/net/tcp"),
+                "  sl  local_address rem_address   st tx_queue rx_queue tr tm->when retrnsmt   uid  timeout inode\r\n" +
```

Comment:
Any reason not to use verbatim string (which means we don't need to concat or format substrings)?
E.g.,
```cs
 @$"  sl  local_address rem_address   st tx_queue rx_queue tr tm->when retrnsmt   uid  timeout inode
   0: 030011AC:8AF2 C1B17822:01BB 01 00000000:00000000 02:000000D1 00000000   472        0 2481276 2 00000000c62511cb 28 4 26 10 -1
"
```


Also, have you considered moving the blob text into a separate files, which then get loaded during the test? In the external files we could also use Linux line endings (i.e., `\n` instead of `\r\n`).

Reply:
> moving the blob text into a separate files

Can do that, as long as it improves code readability (which is not the case for me, I'd rather read the code without switching context to other files). Let me know